normandy/recipes/models.py

```diff
@@ -449,18 +450,24 @@ def _create_new_enabled_state(self, **kwargs):
         self.recipe.update_signature()
         self.recipe.save()
 
+    @transaction.atomic
     def enable(self, user, carryover_from=None):
         if self.enabled:
             raise EnabledState.NotActionable("This revision is already enabled.")
 
         self._create_new_enabled_state(creator=user, enabled=True, carryover_from=carryover_from)
 
+        RemoteSettings().publish(self.recipe)
```

Comment:
Why this? That means the `self.client` class instance object needs to be re-created for every recipe. Perhaps that's immaterial since it's so rare and cheap.